bson/src/main/org/bson/codecs/pojo/annotations/BsonProperty.java

```diff
@@ -27,7 +27,8 @@
 /**
  * An annotation that configures a property.
  *
- * <p>Note: Requires the {@link org.bson.codecs.pojo.Conventions#ANNOTATION_CONVENTION}</p>
+ * <p>For POJOs, requires the {@link org.bson.codecs.pojo.Conventions#ANNOTATION_CONVENTION}</p>
+ * <p>For Java records, the annotation is only supported on the record component.</p>
  *
  * @since 3.5
  * @see org.bson.codecs.pojo.Conventions#ANNOTATION_CONVENTION
```

Comment:
The value() method documentation should also be updated in relation to records.

Reply:
I'm not sure what to say on the `value` method.  The explanatory paragraph would only apply to POJOs because there can't be asymmetrical-ness on records.  

Can you make a suggestion so I know what you're looking for?

Reply:
Something like this?

```
    /**
     * The name of the property.
     *
     * <p>
     *     <strong>Note:</strong> Regarding POJOs:<br />
     *     For asymmetrical property names, the context of the {@code BsonProperty} can be important.
     *     For example, when used with {@code @BsonCreator} the value will relate to the read name.
     *     When used directly on a field it will set both the read name if unset and the write name if unset.
     * </p>
     *
     * @return the name to use for the property
     * @see For POJOs see: PropertyModel#getWriteName()
     * @see For POJOs see: PropertyModel#getReadName()
     */
```